tests/sles4sap/hana_install.pm

```diff
@@ -69,19 +69,19 @@ sub run {
     }
     else {
         # If running on QEMU and with a second disk configured, then configure
-        # mountpoints and LVM. Otherwise leave those choices to hdblcm, but
-        # always create mountpoints.
+        # mountpoints and LVM. Otherwise leave those choices to hdblcm. If running
+        # in a different backend, assume sdb exists. Always create mountpoints.
         foreach (keys %mountpts) { assert_script_run "mkdir -p $mountpts{$_}->{mountpt}"; }
-        if (check_var('BACKEND', 'qemu') and get_var('HDDSIZEGB_2')) {
-            my $device = check_var('HDDMODEL', 'scsi-hd') ? '/dev/sdb' : '/dev/vdb';
+        if ((check_var('BACKEND', 'qemu') and get_var('HDDSIZEGB_2')) or !check_var('BACKEND', 'qemu')) {
+            my $device = (check_var('HDDMODEL', 'scsi-hd') or !check_var('BACKEND', 'qemu')) ? '/dev/sdb' : '/dev/vdb';
```

Comment:
In any case (bare metal or qemu), maybe it could be a good idea to check if `$device` exists and fail if not with a understandable message like `Device $device is missing, aborting!`.

WDYT?

Reply:
WDYT of a `assert_script_run "ls $device"` below L76? It'll fail the test if the device doesn't exist.

Reply:
Thinking more about this, if the device doesn't exist, module will fail in L78 (parted command) with an `Error: Could not stat device /dev/sdb - No such file or directory` message. So I would be against adding extra code to check for the device existence only for the test to fail. We could check for the device existence and choose to skip the partitioning but continue with the module execution instead, but I think in such cases `sles4sap/hana_test` will fail. Thoughts?

Reply:
Good point, L78 will fail the test with an explicit message in the log file, so forget my comment and let the test as-is.
Note for me: read more carefully the next time :D